tests/numeric_test.py

```diff
@@ -197,21 +197,335 @@ def testCumProd(self):
             ak.cumprod([range(0, 10)])
 
     def testSin(self):
-        na = np.linspace(1, 10, 10)
+        na = np.linspace(1, 10, 10, "int64")
+        pda = ak.array(na)
+        self.assertTrue(np.allclose(np.sin(na), ak.sin(pda).to_ndarray()))
+
+        na = np.linspace(1, 10, 10, "uint64")
         pda = ak.array(na)
+        self.assertTrue(np.allclose(np.sin(na), ak.sin(pda).to_ndarray()))
 
+        na = np.linspace(1, 10, 10, "float64")
+        pda = ak.array(na)
         self.assertTrue(np.allclose(np.sin(na), ak.sin(pda).to_ndarray()))
+
         with self.assertRaises(TypeError):
-            ak.cos([range(0, 10)])
+            ak.sin([range(0, 10)])
 
     def testCos(self):
-        na = np.linspace(1, 10, 10)
+        na = np.linspace(1, 10, 10, "int64")
```

Comment:
this isn't giving u what u want, you have to use the `dtype` keyword to set this argument like this
```python
>>> np.linspace(1, 10, 10, "int64")
array([ 1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9., 10.])

>>> np.linspace(1, 10, 10, "int64").dtype
dtype('float64')

>>> np.linspace(1, 10, 10, dtype="int64").dtype
dtype('int64')
```

also it might just make more sense to use `arange` since i think most ppl are more used to working with that
```python
>>>  np.arange(1, 10)
array([1, 2, 3, 4, 5, 6, 7, 8, 9])

>>> np.arange(1, 10).dtype
dtype('int64')

>>> np.arange(1, 10, dtype=np.uint).dtype
dtype('uint64')

>>> np.arange(1, 10, dtype=float).dtype
dtype('float64')
```

But as it stands right now, the tests are only testing the `float` case

Reply:
Good catch. I matched the format of another test, so assumed this was how the kwargs detected it. Went back in fixed that test too.